R/sidebar_layout.R

```diff
@@ -0,0 +1,208 @@
+# ------------------------------------------------------------------------------
+# Sidebar layout composed of main and sidebar panels
+# ------------------------------------------------------------------------------
+#'
+#' Creates row wrapping specified elements
+#' @param arg Element to be wrapped in row container
+#' @return Row wrapping specified elements
+
+get_row <- function(arg) {
+  class <- "row"
+  style <- "padding: 20px;"
+  HTML(glue::glue("<div class='{class}' style='{style}'>{arg}</div>"))
+}
+
+#' Creates div containing elements in order or grid container for specific panels
+#'
+#' @param grid_list List to create grid from
+#' @param ... Container's children elements
+#' @return Div containing elements or grid container for specific panel
+
+panel <- function(grid_list, ...) {
+  args <- list(...)
+  if(is.null(grid_list)) {
+    div(lapply(args, get_row))
+  } else {
+    grid(grid_list$layout, grid_list$container_style, grid_list$area_styles, ...)
+  }
+}
+
+#' Creates div containing elements in order or grid container for sidebar panel
+#'
+#' @param grid_list List to create grid from
+#' @param ... Container's children elements
+#' @export
+
+sidebar_panel <- function(grid_list, ...) {
+  panel(grid_list, ...)
+  # adjustments for sidebar panel
+}
+
+#' Creates div containing elements in order or grid container for main panel
+#'
+#' @param grid_list List to create grid from
+#' @param ... Container's children elements
+#' @export
+
+main_panel <- function(grid_list, ...) {
+  panel(grid_list, ...)
+  # adjustments for main panel
+}
+
+#' Creates grid layout composed of sidebar and main panels
+#'
+#' @param sidebar_panel Sidebar panel component
+#' @param main_panel Main panel component
+#' @param sidebar_width Width of sidebar panel in percents
+#' @param mirrored If TRUE sidebar is located on the right side,
+#' if FALSE - on the left side (default)
+#' @param min_height Sidebar layout container keeps the minimum height, if
+#' specified. It should be formatted as a string with css units
+#' @param container_style CSS declarations for grid container
+#' @param area_styles List of CSS declarations for each grid area inside container
+#'
+#' @return Container with sidebar and main panels
+#' @examples
+#'
+#' sidebar_layout(
```

Comment:
I really like the example from the `examples/` but the one here should demonstate a simple usecase, eg reproduce an example from `?shiny::sidebarLayout()`